alchemist-euclidean-geometry/src/test/java/it/unibo/alchemist/test/TestContinuous2DObstacle.java

```diff
@@ -25,39 +28,57 @@
     private static final RectObstacle2D<Euclidean2DPosition> R1021 = new RectObstacle2D<>(1, 0, 1, 1);
     private static final RectObstacle2D<Euclidean2DPosition> R0527 = new RectObstacle2D<>(0, 5, 2, -2);
 
+    private Node<Integer> createIntNode(
+            final Incarnation<Integer, Euclidean2DPosition> incarnation,
+            final Environment<Integer, Euclidean2DPosition> environment
+    ) {
+        return new GenericNode<>(incarnation, environment) {
+            @Override
+            public Integer createT() {
+                return 0;
+            }
+        };
+    }
+
     /**
      * 
      */
     @Test
     void test() {
         final var incarnation = SupportedIncarnations.<Integer, Euclidean2DPosition>get("protelis").orElseThrow();
-        final Continuous2DObstacles<Integer> env = new Continuous2DObstacles<>(incarnation);
-        env.setLinkingRule(new NoLinks<>());
-        env.addObstacle(R1021);
-        env.addObstacle(R0527);
-
-        assertEquals(new Euclidean2DPosition(FastMath.nextAfter(1.0, 0.0), FastMath.nextAfter(1.0, 0.0)), env.next(0, 0, 1, 1));
-        assertEquals(new Euclidean2DPosition(0, 0), env.next(1, 1, 0, 0));
-        assertEquals(new Euclidean2DPosition(FastMath.nextAfter(1.0, 0.0), FastMath.nextAfter(0.5, 0.0)), env.next(0, 0, 2, 1));
-
-        env.addNode(new IntNode(env), new Euclidean2DPosition(0, 0));
-        assertEquals(env.getNodeCount(), 1);
-        env.addNode(new IntNode(env), new Euclidean2DPosition(1, 1));
-        assertEquals(env.getNodeCount(), 1);
+        final Continuous2DObstacles<Integer> environment = new Continuous2DObstacles<>(incarnation);
+        environment.setLinkingRule(new NoLinks<>());
+        environment.addObstacle(R1021);
+        environment.addObstacle(R0527);
+
+        assertEquals(new Euclidean2DPosition(
+                FastMath.nextAfter(1.0, 0.0),
+                FastMath.nextAfter(1.0, 0.0)
+        ), environment.next(0, 0, 1, 1));
```

Comment:
```suggestion
        assertEquals(
            new Euclidean2DPosition(FastMath.nextAfter(1.0, 0.0), FastMath.nextAfter(1.0, 0.0)),
            environment.next(0, 0, 1, 1)
        );
```

also static import `nextAfter`